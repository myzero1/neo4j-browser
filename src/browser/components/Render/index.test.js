/*
 * Copyright (c) 2002-2019 "Neo4j,"
 * Neo4j Sweden AB [http://neo4j.com]
 *
 * This file is part of Neo4j.
 *
 * Neo4j is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

/* global describe, test, expect */

import React from 'react'
import { render } from 'react-testing-library'

import Render from './index'

describe('<Render>', () => {
  test('does not render if condition is false', () => {
    // Given
    const val = false

    // When
    const { container } = render(
      <Render if={val}>
        <span>Hello</span>
      </Render>
    )

    // Then
    expect(container).toMatchSnapshot()
  })
  test('does render if condition is true', () => {
    // Given
    const val = true

    // When
    const { container } = render(
      <Render if={val}>
        <span>Hello</span>
      </Render>
    )

    // Then
    expect(container).toMatchSnapshot()
  })
})
