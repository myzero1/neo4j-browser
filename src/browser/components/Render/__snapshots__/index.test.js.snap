// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Render> does not render if condition is false 1`] = `<div />`;

exports[`<Render> does render if condition is true 1`] = `
<div>
  <span>
    Hello
  </span>
</div>
`;
